### ISSUE TITLE
[Android] `onBlur` closes the keyboard but doesn't remove focus on input

### ISSUE THREAD

**Issue by kperdomo1** [stripe/stripe-react-native#429]
**Describe the bug**
On `v0.1.5`'s `CardField`, when executing the imperative `onBlur` method on Android it closes the keyboard successfully, but focus moves to the card number input.

The same behaviour also applies when clicking anywhere on the parent component. Keyboard closes, but focus moves to the card number input.

**It works perfectly for iOS**

**To Reproduce**
1. Use CardField component.
2. Provide all card details
3. Execute the imperative `onBlur` method.
4. Keyboard closes but focus moves to the card number input.

**Expected behavior**
When keyboard closes, no other input should be focused.

**Screenshots**

https://user-images.githubusercontent.com/8103420/125928646-5e464535-201c-49a8-a055-343ebae1eed3.mov

**Smartphone (please complete the following information):**
 - Device: Android Pixel 3 emulator
 - OS: Android 11

Also happens on physical device/
- Device: Samsung S10+
- OS: Android 11

**Comment by thorsten-stripe** [comment]
That's unfortunately a known limitation on Android currently.

**Comment by kperdomo1** [comment]
I see, thanks @thorsten-stripe . 
FYI, keeping the `<StripeContainer keyboardShouldPersistTaps={false}>` and `<ScrollView keyboardShouldPersistTaps="always">` fixes this behaviour for Android when clicking outside the input.

The imperative onBlur issue still persists though.

**Comment by tantommy** [comment]
@kperdomo1 do you happen to have any issues with iOS? When i tap away from the `CardField` after focusing on it, i'm seeing this error
```
No command found with name "blur"
ABI42_0_0facebook::ABI42_0_0React::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
    ABI42_0_0RCTUIManager.m:1125
4FBDF167-161A-324C-A233-D516922C67E5
4FBDF167-161A-324C-A233-D516922C67E5
4FBDF167-161A-324C-A233-D516922C67E5
ABI42_0_0facebook::ABI42_0_0React::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
ABI42_0_0facebook::ABI42_0_0React::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
ABI42_0_0facebook::ABI42_0_0React::JSIExecutor::defaultTimeoutInvoker(std::__1::function<void ()> const&, std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > ()>)
DAF30062-4C85-3B92-B159-50602A0C9D97
DAF30062-4C85-3B92-B159-50602A0C9D97
DAF30062-4C85-3B92-B159-50602A0C9D97
DAF30062-4C85-3B92-B159-50602A0C9D97
DAF30062-4C85-3B92-B159-50602A0C9D97
_pthread_wqthread
start_wqthread
```
Once I add `<ScrollView keyboardShouldPersistTaps="always">`, that doesn't happen anymore but then it happens again when i navigate away to another screen.  The app just crashes with the error above.

**Comment by kperdomo1** [comment]
I haven't faced any issue on iOS @tantommy , maybe you can create a more detailed issue? could be a separate problem.
The `onBlur` call works for me in both platforms, just passing down the handler to the onBlur prop.

**Comment by tantommy** [comment]
> I haven't faced any issue on iOS @tantommy , maybe you can create a more detailed issue? could be a separate problem.
> The `onBlur` call works for me in both platforms, just passing down the handler to the onBlur prop.

I've found out that when I run the app in the simulator, everything works fine.  the onBlur is called and the error i posted above is gone.  Something about the Expo Go development env is giving me the error above.

**Comment by thorsten-stripe** [comment]
Each Expo SDK version requires a specific `stripe-react-native` version. See the [CHANGELOG](./CHANGELOG.md) for a mapping of versions. To install the correct version for your Expo SDK version run:

```sh
expo install @stripe/stripe-react-native
```

**Comment by lucasroca** [comment]
Hi @thorsten-stripe @kperdomo1 , do you know of any workaround to this issue? I have a CardField and a button below. When clicking outside the CardField, the keyboard is dismissed, but the input remains focused, and the button doesn't work.

**Comment by kperdomo1** [comment]
@lucasisola did you wrap both the CardField and button inside the ScrollView ?

**Comment by lucasroca** [comment]
@kperdomo1, I removed the styling, this is what I'm trying:

```javascript
const InputCreditCard = props => {
  useEffect(() => {
    initStripe({ publishableKey: props.stripeKey });
  }, []);

  const onSubmit = () => console.log('Submit');

  return (
    <ScrollView
      contentContainerStyle={{ paddingHorizontal: 16 }}
      showsVerticalScrollIndicator={false}
      keyboardShouldPersistTaps="never"
    >
      <CardField postalCodeEnabled={false} autofocus style={{ height: 50, marginVertical: 30 }} />
      <Button onPress={onSubmit} title="Pay" />
    </ScrollView>
  );
};
```

https://user-images.githubusercontent.com/29613530/127329180-dcee0f9e-7e1e-435d-8516-60ed004b4528.mov

I see you mentioned `StripeContainer`, which I am not using. Could that be the problem?

**Comment by arekkubaczkowski** [comment]
> Hi @thorsten-stripe @kperdomo1 , do you know of any workaround to this issue? I have a CardField and a button below. When clicking outside the CardField, the keyboard is dismissed, but the input remains focused, and the button doesn't work.

@lucasroca I am able to reproduce it within react-native 062.2 version, could you create a separate issue?

**Comment by arekkubaczkowski** [comment]
@kperdomo1 this is gonna be fixed in the next release (0.1.6)
@lucasroca your issue same

**Comment by lucasroca** [comment]
Great! Also, I found out my button wasn't clickable because I was using nested ScrollViews like:

```javascript

<ScrollView ...>
...
  <InputCreditCard />
...
</ScrollView>
```
InputCreditCard being the component from my previous comment. So I removed one ScrollView and now it's working.

Thanks for the support!

**Comment by babyrusa** [comment]
I was having this issue. I didn't use `StripeContainer` / have not tried but I notice not having ScrollView like @lucasisola  said helped resolving it.

**Comment by arekkubaczkowski** [comment]
it's already fixed in `0.2.0v`
